src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Todo;
+import duke.task.Event;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+/**
+ * Contains the processes that store data and manipulates the data file of the application
+ */
+public class Storage {
+    /** list that stores the tasks */
+    private ArrayList<Task> list = new ArrayList<>();
+
+    /** Keeps track of the number of tasks */
+    private int count = 0;
+
+    public Storage() {
+        try {
+            if (!new File("data").exists()) {
+                new File("data").mkdirs();
+                File tmp = new File("data", "duke.txt");
+                tmp.createNewFile();
+            } else if (!new File("data/duke.txt").exists()) {
+                File tmp = new File("data", "duke.txt");
+                tmp.createNewFile();
+            } else {
+                this.loadData();
+            }
+        } catch(IOException e){
+            System.out.println("Something went wrong");
+        }
+    };
+
+    /**
+     * loads the data into the list from the data file
+     */
+    public void loadData() {
+        try {
+            File f = new File("data/duke.txt"); // create a File for the given file path
+            Scanner s = new Scanner(f); // create a Scanner using the File as the source
+
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String[] parts = line.split("##");
+                switch(parts[0]) {
+                    case("[E]"):
```

Comment:
Maybe you could remove the extra indentation in the case statement, as indicated in the Java coding standard